src/services/hooks/useUserFeedback.ts

```diff
@@ -0,0 +1,19 @@
+import { useState } from "react";
+
+export const useUserFeedback = () => {
+  const [showSuccess, setShowSuccess] = useState<boolean>(false);
+  const [showError, setShowError] = useState<boolean>(false);
+  const [errorHeading, setErrorHeading] = useState<string | null>(null);
+  const [errorDescription, setErrorDescription] = useState<string | null>(null);
+
+  return {
+    showSuccess,
+    setShowSuccess,
+    showError,
+    setShowError,
+    errorHeading,
+    setErrorHeading,
+    errorDescription,
+    setErrorDescription,
+  };
+};
```

Comment:
Is there any common functionality you could group together, such as clearForm, setSuccess etc? Custom hooks are good for encapsulating complexity. Whereas here, it's being used to group various states together. 

If you cant implement the logic within the hook itself, it's probably adding unnecessary complexity and will be more readable to directly implement these state values wherever the hook is used.

Reply:
I just wanted to be cool and create a custom hook but I see your point. It removed a little repeated code but not enough to justify it. Removing the hook now from the 2 components that were implementing it.